### PR TITLE
feat(http): expose connection manager

### PR DESCRIPTION
**Pull request by dargmuesli**
Builds upon https://github.com/spotify-web-api-java/spotify-web-api-java/pull/407, keeping the setter-only builder architecture.

### CODE REVIEW

File: src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java [hunk 1]
```diff
@@ -128,6 +135,10 @@ public static URI makeUri(String uriString) {
     }
   }
 
+  public HttpClientConnectionManager getConnectionManager() {
```
Comment (bruceadowns) [inline src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java:138]: getConnectionManager is no longer necessary

File: src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java [hunk 2]
```diff
@@ -75,14 +77,19 @@ public SpotifyHttpManager(Builder builder) {
...
+      basicHttpClientConnectionManager.setConnectionConfig(ConnectionConfig
+        .custom()
+        .setConnectTimeout(connectTimeout != null ?
+          Timeout.ofMilliseconds(connectTimeout) :
+          ConnectionConfig.DEFAULT.getConnectTimeout())
+        .build());
+      this.connectionManager = basicHttpClientConnectionManager;
+    } else {
+      this.connectionManager = builder.connectionManager;
+    }
```
Comment (bruceadowns) [inline src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java:91]: nitpick consider conditional consistency throughout the library.

either this:
```
    if (value == null) {
      // something
    } else {
      // something else
    }
```

or this:
```
    if (value != null) {
      // something
    } else {
      // something else
    }
```

**Review by bruceadowns — APPROVED** [review]
lgtm, I like keeping the builder usage consistent.

File: src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java (hunk 1 above)
Comment (dargmuesli) [reply src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java:138]: We have those getters for all class variables right now, like `getProxy`, ... which we don't use in our code base either, so I just added this as well.

File: src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java (hunk 2 above)
Comment (dargmuesli) [reply src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java]: Switched :white_check_mark:

**Pull request merged**